include/cufinufft/impl.h

```diff
@@ -354,6 +354,115 @@ Notes: the type T means either single or double, matching the
     return ier;
 }
 
+template<typename T>
+T calculate_scale_factor(int rank, finufft_spread_opts opts) 
```

Comment:
We have no idea what this code does without a mathematical description, and documentation of inputs/outputs.